Internal documentation [14071] (#161)
eProsima/DDS-Router: .dev/scripts/README.md

1. Every package must be contained in its directory and every file needed for it must be inside as well.
1. There could not be cyclic dependencies between packages.
1. Every CMake package is configured by a `project_settings.cmake` and should try to use the generic CMakeLists.

### Project settings

There is a special file in each of the packages called `project_settings.cmake`.
This CMake file configure the specific values required to parametrize the common `CMakeLists` that are used in every
project.
The values that must be initialized are:

* `SUBMODULE_PROJECT_NAME` project name
* `SUBMODULE_PROJECT_SUMMARY` project summary
* `SUBMODULE_PROJECT_FIND_PACKAGES` packages that the project needs to find by `find_package` cmake command
* `SUBMODULE_PROJECT_DEPENDENCIES` project dependencies (usually: `${SUBMODULE_PROJECT_FIND_PACKAGES}`)
* `SUBMODULE_PROJECT_MACROS` project macros name (usually: `$SUBMODULE_PROJECT_NAME` in capital letters). This is used for example to set the dll macro: ${SUBMODULE_PROJECT_MACROS}_DllAPI

---

## Design decisions

### License

Every file must have its license at its head.

> This license must concordance with this one (change the year to the year the file has been created):

Reviewer: ```suggestion
This license be consistent with the following one (change the year to the year the file has been created):
```
